### PR TITLE
Add troubleshoot article for asset caching

### PR DESCRIPTION
**Pull request by vicroms**

### CODE REVIEW

File: vcpkg/troubleshoot/asset-caching.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+Containes and other storage servies that accept requests that require an
+authentication token.
+
+The configuration is as follows:
+
+`x-azurl,<url>,<sas>[,<rw>]`
+
+* `<url>`: The container's base URL
+* `<sas>`: A Shared Access Signature (SAS) token if working with Azure Storage
+  Contianers; or an authentication request parameter if working with other providers.
```
Comment (AugP) [inline vcpkg/troubleshoot/asset-caching.md:37]:
```suggestion
  Containers, or an authentication request parameter if working with other providers.
```

File: vcpkg/troubleshoot/asset-caching.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+up](../commands/install.md) an asset cache. For instructions to set up an asset
+cache read the [asset caching documentation](../concepts/asset-caching.md).
+
+## Artifacts aren't being uploaded or restored from my asset cache
+
+When vcpkg downloads assets it looks at available asset caches before
+attempting to download from the Internet.
+
+By default, vcpkg emits no warnings or errors when a package is not found on an
+asset cache. Instead, it silently falls back to downloading the asset from the Internet.
```
Comment (vicroms) [inline vcpkg/troubleshoot/asset-caching.md:22]:
```suggestion
By default, vcpkg emits no warnings or errors when a package is not found on an
asset cache. Use the [`x-block-origin`](../users/assetcaching.md#x-block-origin) to
disable the silent download mechanism, turning a failure to restore from an asset
cache into a package build failure.
```

File: vcpkg/troubleshoot/asset-caching.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+`x-azurl,https://contoso.com,authorization=tokenvaluehere,readwrite`.
+
+Steps to resolve:
+
+1 - Verify that the authorization token is not expired
+2 - Verify that the authorization token has the correct permissions to your container
+3 - Verify that the authorization token's value is correct
+4 - Verify that the authorization token's format matches the expected format by
+your provider.
```
Comment (JavierMatosD) [inline vcpkg/troubleshoot/asset-caching.md:97]: These aren't rendering as a numbered list:

image

File: vcpkg/troubleshoot/asset-caching.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+ms.author: viromer
+ms.date: 02/12/2024
+ms.topic: troubleshooting-general
+---
+
+# Troubleshoot asset caching issues
+
+This article is intended for users experiencing issues while setting
+up an asset cache. For instructions to set up an asset cache read the
+[asset caching documentation](../concepts/asset-caching.md).
```
Comment (JavierMatosD) [inline vcpkg/troubleshoot/asset-caching.md:14]: Maybe a brief description before jumping in?


AI suggested:
Asset caching accelerates package retrieval by storing downloaded packages (assets) locally or on a network, reducing dependency on external sources. This guide assists users facing difficulties with asset cache setup or operation. For initial setup instructions, refer to the [Asset Caching Documentation](../concepts/asset-caching.md)

File: vcpkg/troubleshoot/asset-caching.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+failures to restore assets into package build failures.
+
+vcpkg prints the URL for each of the assets it downloads. You can detect asset
+caching errors if the download URL does not match your asset cache's URL. If
+you added `x-block-origin`, failing to restore an asset from one of the caches
+fails with an error message:
+
+```Console
+error: Failed to download from mirror set
+```
```
Comment (JavierMatosD) [inline vcpkg/troubleshoot/asset-caching.md:33]:
```suggestion
vcpkg checks available asset caches before downloading artifacts from the Internet. By default, it silently falls back to external sources if an asset is not found in the cache. 

To enhance security and make missing cache hits more visible, add the [x-block-origin](../users/assetcaching.md#x-block-origin) option to your asset cache configuration. This option transforms asset restoration failures into explicit build errors, preventing unintended external downloads.

Indications of Asset Caching Issues:
* Downloads not utilizing cache URLs.
* The presence of `x-block-origin` results in explicit errors like:

```console
error: Failed to download from mirror set
```
```

File: vcpkg/troubleshoot/asset-caching.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+The configuration has the following format:
+
+`x-azurl,<url>,<sas>[,<rw>]`
+
+* `<url>`: The container's base URL
+* `<sas>`: A Shared Access Signature (SAS) token if working with Azure Storage
+  Containers; or an authentication request parameter if working with other providers.
+* `<rw>`: (Optional) read/write permission configuration
+
+> !NOTE
```
Comment (JavierMatosD) [inline vcpkg/troubleshoot/asset-caching.md:50]:
```suggestion
> [!NOTE]
```

This is how it's currently rendering:
image

File: vcpkg/troubleshoot/asset-caching.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+
+Steps to resolve:
+
+1 - Verify that the authorization token is not expired
+2 - Verify that the authorization token has the correct permissions to your container
+3 - Verify that the authorization token's value is correct
+4 - Verify that the authorization token's format matches the expected format by
+your provider.
+
+## I get a "Failed to download from mirror set" error message
```
Comment (JavierMatosD) [inline vcpkg/troubleshoot/asset-caching.md:99]: I'm not 100% sure, but I think you can get this error message if there is a network issue or the mirror is down for some reason. Maybe add that as a set to resolve?

**Review by JavierMatosD — CHANGES_REQUESTED** [review]

**Pull request merged**